packages/cubejs-server-core/src/core/CompilerApi.js

```diff
@@ -229,22 +423,102 @@ export class CompilerApi {
     );
   }
 
-  async metaConfig(options = {}) {
+  /**
+   * if RBAC is enabled, this method is used to filter out the cubes that the
+   * user doesn't have access from meta responses.
+   * It evaluates all applicable memeberLevel accessPolicies givean a context
+   * and retains members that are allowed by any policy (most permissive set).
+  */
+  async filterVisibilityByAccessPolicy(cubeEvaluator, context, cubes) {
+    const isMemberVisibleInContext = {};
+
+    if (!cubeEvaluator.isRbacEnabled()) {
+      return cubes;
+    }
+
+    for (const cube of cubes) {
+      const evaluatedCube = cubeEvaluator.cubeFromPath(cube.config.name);
+      if (cubeEvaluator.isRbacEnabledForCube(evaluatedCube)) {
+        const applicablePolicies = await this.getApplicablePolicies(evaluatedCube, context, cubeEvaluator);
+
+        const computeMemberVisibility = async (item) => {
+          let isIncluded = false;
+          let isExplicitlyExcluded = false;
+          for (const policy of applicablePolicies) {
+            if (policy.memberLevel) {
+              isIncluded = policy.memberLevel.includesMembers.includes(item.name) || isIncluded;
+              isExplicitlyExcluded = policy.memberLevel.excludesMembers.includes(item.name) || isExplicitlyExcluded;
+            } else {
+              isIncluded = true;
+            }
+          }
+          return isIncluded && !isExplicitlyExcluded;
+        };
+
+        for (const dimension of cube.config.dimensions) {
+          isMemberVisibleInContext[dimension.name] = await computeMemberVisibility(dimension);
+        }
+
+        for (const measure of cube.config.measures) {
+          isMemberVisibleInContext[measure.name] = await computeMemberVisibility(measure);
+        }
+
+        for (const segment of cube.config.segments) {
+          isMemberVisibleInContext[segment.name] = await computeMemberVisibility(segment);
+        }
+      }
+    }
+
+    const visibilityFilterForCube = (cube) => {
+      const evaluatedCube = cubeEvaluator.cubeFromPath(cube.config.name);
+      if (!cubeEvaluator.isRbacEnabledForCube(evaluatedCube)) {
+        return (item) => item.isVisible;
+      }
+      return (item) => (item.isVisible && isMemberVisibleInContext[item.name] || false);
+    };
+
+    return cubes
+      .map((cube) => ({
+        config: {
+          ...cube.config,
+          measures: cube.config.measures?.filter(visibilityFilterForCube(cube)),
+          dimensions: cube.config.dimensions?.filter(visibilityFilterForCube(cube)),
+          segments: cube.config.segments?.filter(visibilityFilterForCube(cube)),
+        },
+      })).filter(
+        cube => cube.config.measures?.length ||
+          cube.config.dimensions?.length ||
+          cube.config.segments?.length
+      );
+  }
+
+  async metaConfig(requestContext, options = {}) {
```

Comment:
Maybe put `requestContext` also in `options`?

Reply:
It's always required